tests/integration/tests/test_cloudbeat_integration.py

```diff
@@ -55,6 +55,55 @@ def test_elastic_index_exists(elastic_client, match_type):
     :param match_type: Findings type for matching
     :return:
     """
+    query, sort = build_es_query(match_type)
+    start_time = time.time()
+    result = {}
+    while time.time() - start_time < CONFIG_TIMEOUT:
+        current_result = elastic_client.get_index_data(index_name=elastic_client.index,
+                                                       query=query,
+                                                       sort=sort)
+        if elastic_client.get_total_value(data=current_result) != 0:
+            allure.attach(json.dumps(elastic_client.get_doc_source(data=current_result),
+                                     indent=4,
+                                     sort_keys=True),
+                          match_type,
+                          attachment_type=allure.attachment_type.JSON)
+            result = current_result
+            break
+        time.sleep(1)
+
+    assert len(result) > 0, \
+        f"The findings of type {match_type} not found"
+
+
+@pytest.mark.pre_merge
+@pytest.mark.order(4)
+@pytest.mark.dependency(depends=["test_cloudbeat_pod_exist"])
+def test_leader_election(fixture_data, elastic_client):
+    """
+    This test verifies that findings of all types are sending to elasticsearch
+    :param elastic_client: Elastic API client
+    :return:
+    """
+
+    query, sort = build_es_query("k8s_object")
+    pods, nodes, leader_node = fixture_data
+    assert leader_node != "", \
+        f"The Leader node could not be found"
+
+    time.sleep(CONFIG_TIMEOUT)  # Wait for all pods to send resources to ES
+
+    result = elastic_client.get_index_data(index_name=elastic_client.index,
+                                           query=query,
+                                           size=1000,
+                                           sort=sort)
+    # checking that k8s_objects are being sent only by the leader node.
+    for resource in result['hits']['hits']:
+        assert leader_node == resource['_source']['agent']['name'], \
+            f"Multiple agents send k8s_objects"
```

Comment:
Why did you put assert inside for loop? Assert will break execution of 'for' loop on failure with a common message and you will not know where is the issue.
So if by design the test shall be interrupted, please format failure string (line 103) and provide more information using variables.
If you want to continue and iterate over all resources, you need to collect them to a list/dictionary (instead of assert), and then to perform assert once to check if a list/dictionary is empty.

Reply:
I'm fine with stopping the test execution, I'll make the log more comprehensive.